### PR TITLE
Add new syntax for NONIUS_BENCHMARK

### PR DESCRIPTION
**Pull request by arximboldi**
Addresses #10

### CODE REVIEW

**Comment by rmartinho** [comment]
The build server cannot build this PR because the author is not whitelisted. A meatbag will have to look at it.

**This is an automated message.**

File: include/nonius/benchmark.h++ [hunk 1]
```diff
@@ -83,15 +83,39 @@ namespace nonius {
...
+    }                                                                   \
+    //
+
+#define NONIUS_BENCHMARK_NEW_(name)                                     \
+    static auto NONIUS_DETAIL_UNIQUE_NAME(benchmark_registrar) =        \
+        ::nonius::benchmark_registrar{::nonius::global_benchmark_registry(), name} + \
+        //
+
+#define NONIUS_BENCHMARK_CHOOSE_(_1,_2,_3,_4,_5,_6,_7,_8,_9,_10,_11,_12,_13,_14,_15,_16,_17,_18,_19,_20,_21,_22,_23,_24,_25,_26,_27,_28,_29,_30,_31,_32,_33,_34,_35,_36,_37,_38,_39,_40,_41,_42,_43,_44,_45,_46,_47,_48,_49,_50,_51,_52,_53,_54,_55,_56,_57,_58,_59,_60,_61,_62,_63,_64,_65,_66,_67,_68,_69,_70,_71,_72,_73,_74,_75,_76,_77,_78,_79,_80,_81,_82,_83,_84,_85,_86,_87,_88,_89,_90,_91,_92,_93,_94,_95,_96,_97,_98,_99,_100,_101,_102,_103,_104,_105,_106,_107,_108,_109,_110,_111,_112,_113,_114,_115,_116,_117,_118,_119,_120,_121,_122,_123,_124,_125,_126,_127,_128,N,...) N
+#define NONIUS_BENCHMARK_CAT__(X1, X2) X1 ## X2
```
Comment (rmartinho) [inline include/nonius/benchmark.h++:115]: Oh damn, I missed this. Double underscores in identifiers are reserved EVERYWHERE.

Comment (arximboldi) [reply include/nonius/benchmark.h++:115]: Sorry I missed that too... I'll change it.

Comment (arximboldi) [reply include/nonius/benchmark.h++:115]: Done.

**Comment by arximboldi** [comment]
Oh shit, I missed that you merged this, I am going to remake this on top
of current `devel`.

**Comment by arximboldi** [comment]
Done, again :-)

**Comment by rmartinho** [comment]
No worries. I wanted to make a beta release from the current PRs and I kinda messed up with this one because I thought I had gone over it before, which I now doubt I did :D (I even got the docs wrong :S). I also realised I will want to deal with deprecation and backwards-compatibility (#68) before this gets shipped.

**Comment by rmartinho** [comment]
For now I reverted this :/ GitHub doesn't seem to have a way to mark a PR as unmerged, though.

File: include/nonius/benchmark.h++ (hunk 1 above)
Comment (nabijaczleweli) [reply include/nonius/benchmark.h++:115]: @rmartinho not only at ident start?

**Comment by rmartinho** [comment]
Doubles are reserved everywhere, no exceptions.

**Comment by arximboldi** [comment]
Yeah, I see that the pull request did also not updated with my latest push to this branch. I am going to re-open the PR so we can discuss further improvements there.

**Pull request merged**